### PR TITLE
ENT-2721 | Added extra exception handling for discovery call and Ente…

### PR DESCRIPTION
**Pull request by hasnain-naveed**
…rpriseCustomerUser model.

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
- [ ] Version pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)

### CODE REVIEW

File: enterprise/management/commands/create_enterprise_course_enrollments.py [hunk 1]
```diff
@@ -39,50 +42,81 @@ def add_arguments(self, parser):
...
+            enterprise_customer_uuid = item['enterprise_customer_uuid']
 
             # pylint: disable=no-member
             enterprise_customer = EnterpriseCustomer.objects.get(uuid=enterprise_customer_uuid)
-            if enterprise_customer.catalog_contains_course(course_run_id):
-                enterprise_customer_user = EnterpriseCustomerUser.objects.get(
+
+            try:
+                does_course_exist_in_catalog = enterprise_customer.catalog_contains_course(course_run_id)
+            except (SlumberBaseException, ConnectionError, Timeout) as exc:
```
Comment (ziafazal) [inline enterprise/management/commands/create_enterprise_course_enrollments.py:61]: This exception handling needs to done [here](https://github.com/edx/edx-enterprise/blob/dc274195f43fb118c1b7482acc94e5b6fe4c9020/enterprise/models.py#L1536) to cover larger use cases

File: enterprise/management/commands/create_enterprise_course_enrollments.py [hunk 2]
```diff
@@ -108,7 +142,8 @@ def _fetch_course_enrollment_data(self, enterprise_customer_uuid):
                 ON ece.enterprise_customer_user_id = ecu.id
                 AND ece.course_id = sce.course_id
             WHERE
-                ece.id IS NULL
+                ecu.linked = true
```
Comment (ziafazal) [inline enterprise/management/commands/create_enterprise_course_enrollments.py:145]: why we need this extra condition `linked = true`?

File: enterprise/models.py [hunk 3]
```diff
@@ -932,6 +932,15 @@ def create_order_for_enrollment(self, course_run_id, discount_percentage, sales_
         else:
             LOGGER.warning(failure_log_msg, self.user_id, course_run_id, 'Failed to retrieve a valid ecommerce worker.')
 
+    @classmethod
+    def inactive_users_enterprise_customers(cls, user_id, enterprise_customer):
```
Comment (ziafazal) [inline enterprise/models.py:936]: can we rename this method to `inactivate_other_customers`?

File: enterprise/management/commands/create_enterprise_course_enrollments.py (hunk 2 above)
Comment (hasnain-naveed) [reply enterprise/management/commands/create_enterprise_course_enrollments.py:145]: Now, we don't delete the record of the model but we set linked=false. And we need only those records which are linked to enterprise_customers.

**Pull request merged**